engine/schema/src/main/java/com/cloud/upgrade/DatabaseUpgradeChecker.java

```diff
@@ -417,11 +417,12 @@ private void initializeDatabaseEncryptors() {
 
             errorMessage = "Unable to get the 'init' value from 'configuration' table in the 'cloud' database";
             String sql = "SELECT value from configuration WHERE name = 'init'";
-            PreparedStatement pstmt = conn.prepareStatement(sql);
-            ResultSet result = pstmt.executeQuery();
-            if (result.next()) {
-                String init = result.getString(1);
-                s_logger.info("init = " + DBEncryptionUtil.decrypt(init));
+            try (PreparedStatement pstmt = conn.prepareStatement(sql);
+                 ResultSet result = pstmt.executeQuery()) {
+                if (result.next()) {
+                    String init = result.getString(1);
+                    s_logger.info("init = " + DBEncryptionUtil.decrypt(init));
+                }
```

Comment:
this try is good, but it would be better to have it in a separate method. It will be clearer (from the stacktrace) in which try something went wrong. (nested try clause)

Reply:
make sense. done